### PR TITLE
Delete operator implementation

### PR DESCRIPTION
**Pull request by kacper776**
Delete operator implementation.
TLDR: analogical to Insert

### CODE REVIEW

File: komfydb/execution/delete.cc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  int deleted = 0;
+  // XXX In order to call bufferpool only once we
+  // gather (materialise) them using std::vector
+  std::vector<RecordId> ids;
+  ITERATE_RECORDS(child, record) {
+    ids.push_back((*record)->GetId());
+    deleted++;
+  }
+  RETURN_IF_NOT_OOR(record.status());
+  if (deleted == 0) {
```
Comment (korbiniak) [inline komfydb/execution/delete.cc:70]: why not ids.empty()?

File: komfydb/execution/delete.cc [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+void Delete::Explain(std::ostream& os, int indent) {
+  os << Indent(indent) << "-> Deleting from table " << table_id << "\n";
+  child->Explain(os, indent + child_indent);
+}
+
+absl::Status Delete::FetchNext() {
+  int deleted = 0;
+  // XXX In order to call bufferpool only once we
+  // gather (materialise) them using std::vector
```
Comment (korbiniak) [inline komfydb/execution/delete.cc:63]: them -> records

File: komfydb/execution/delete.cc [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      bufferpool(std::move(bufferpool)),
+      tid(tid),
+      table_id(table_id) {}
+
+absl::StatusOr<std::unique_ptr<Delete>> Delete::Create(
+    std::unique_ptr<OpIterator> child, uint32_t table_id,
+    std::shared_ptr<BufferPool> bufferpool, TransactionId tid) {
+  std::vector<Type> types = {Type::INT};
+  std::vector<std::string> fields = {"deleted_tuples"};
+  return std::unique_ptr<Delete>(new Delete(std::move(child), table_id,
```
Comment (korbiniak) [inline komfydb/execution/delete.cc:38]: shouldn't we check if child is from a single table? E.g. return error if this operator joins tuples. If not here, then in bufferpool for sure.

File: komfydb/storage/buffer_pool.cc [hunk 4]
```diff
@@ -105,4 +105,20 @@ absl::Status BufferPool::InsertTuples(
...
+    if (!page || ids[i].GetPageId() != page->GetId()) {
+      ASSIGN_OR_RETURN(
+          page, GetPage(tid, ids[i].GetPageId(), Permissions::READ_WRITE));
+    }
+    RETURN_IF_ERROR(page->RemoveRecord(ids[i]));
+    page->SetDirty(true, tid);
+  }
+  return absl::OkStatus();
+}
+
```
Comment (DorosMichal) [inline komfydb/storage/buffer_pool.cc:123]: Id give more descriptive variable names, ids_to_remove (im not convinced to this one either)

**Review by DorosMichal — APPROVED** [review]

File: komfydb/execution/delete.cc (hunk 3 above)
Comment (kacper776) [reply komfydb/execution/delete.cc:38]: Right now `page->RemoveRecord()` (called by bufferpool) will return error if it ecounters nonsensical RecordId.

Comment (korbiniak) [reply komfydb/execution/delete.cc:39]: Why not return it immediately?

Comment (kacper776) [reply komfydb/execution/delete.cc:39]: How to know if child joins tuples without getting Records from it?

**Pull request merged**